### PR TITLE
Refactor deep tests to separate file, fix deep chunk reads, ripmap reading

### PR DESCRIPTION
**Pull request by kdt3rd**
Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

### CODE REVIEW

File: src/lib/OpenEXRCore/unpack.c [hunk 1]
```diff
@@ -992,6 +995,186 @@ generic_unpack (exr_decode_pipeline_t* decode)
     return EXR_ERR_SUCCESS;
 }
 
+//static exr_result_t
```
Comment (cary-ilm) [inline src/lib/OpenEXRCore/unpack.c]: Should this commented-out code be removed altogether?

File: src/lib/OpenEXRCore/unpack.c [hunk 2]
```diff
@@ -1017,7 +1200,8 @@ internal_exr_match_decode (
         init_cpu_check = 0;
     }
 
-    /* TODO */
+    /* we don't decode deep for now, it is up to the user to do that (for efficiency) */
+    //if (isdeep) return &generic_unpack_deep;
```
Comment (cary-ilm) [inline src/lib/OpenEXRCore/unpack.c]: Should this commented-out line be removed?

**Review by cary-ilm — APPROVED** [review]
LGTM, except for the commented-out code. OK to keep it temporarily if it's useful.

**Comment by kdt3rd** [comment]
I was misinterpreting the data coming in, which is why I had disabled this thinking I would limit scope. But after talking to Peter today, I think I will uncomment it shortly and provide that again

**Pull request merged**